lib/register_common/services/bulk_transformer.rb

```diff
@@ -0,0 +1,56 @@
+require 'register_common/services/file_reader'
+
+require_relative 'msg_handler'
+
+module RegisterCommon
+  module Services
+    class BulkTransformer
+      BATCH_SIZE = 25
+      DEFAULT_PARALLEL_FILES = 1
+      NAMESPACE = 'BULK_TRANSFORMER'.freeze
+
+      def initialize(s3_adapter:, s3_bucket:, set_client:, file_reader: nil, batch_size: nil)
+        @s3_adapter = s3_adapter
+        @s3_bucket = s3_bucket
+        @set_client = set_client
```

Comment:
If I presume correctly, it's possible for this code to be run multiple times, and to use the data written to Redis efficiently on multiple runs, rather than just in-memory whilst it runs?

Reply:
Discussed.